service-base/src/main/java/org/eclipse/hono/service/tenant/TenantHttpEndpoint.java

```diff
@@ -75,21 +79,26 @@ public void addRoutes(final Router router) {
         router.post(path).handler(this::extractRequiredJsonPayload);
         router.post(path).handler(this::checkPayloadForTenantId);
         router.post(path).handler(ctx -> doTenantHttpRequest(ctx, StandardAction.ACTION_ADD,
-                status -> status == HttpURLConnection.HTTP_CREATED));
+                status -> status == HttpURLConnection.HTTP_CREATED,
+                tenantId -> response -> response.putHeader(HttpHeaders.LOCATION,
```

Comment:
IMHO this does make the code very hard to read and, worse, very hard to adapt later. FMPOV it would be much cleaner to have one method per operation/action and then implement them accordingly.

Reply:
I understand your point and will come up with a less condensed version of this code.